File: app/lib/features/fragment/domain/fragment.dart
import 'package:flutter/material.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

import '../../../design/tokens/colors.dart';

part 'fragment.freezed.dart';
part 'fragment.g.dart';

/// Fragment 实体 — "光片"
@freezed
class Fragment with _$Fragment {
  const Fragment._();

  const factory Fragment({
    required int id,
    @Default('') String publicId,
    @Default(0) int userId,
    @Default('') String contentText,
    @Default('说不清') String emotion,
    @Default('twilight') String status,
    @Default([]) List<String> mediaUrls,
    @Default([]) List<String> tags,
    required DateTime createdAt,
    DateTime? updatedAt,
  }) = _Fragment;

  factory Fragment.fromJson(Map<String, dynamic> json) =>
      _$FragmentFromJson(json);

  /// Transitional display helpers. New UI code should use FragmentView instead.
  String get title {
    final firstLine = contentText.trim().split('\n').first;
    if (firstLine.length <= 16) return firstLine;
    return '${firstLine.substring(0, 16)}...';
  }

  String get time {
    final local = createdAt.toLocal();
    return '${local.hour.toString().padLeft(2, '0')}:${local.minute.toString().padLeft(2, '0')}';
  }

  String get dateLabel {
    final local = createdAt.toLocal();
    return '${local.year}年${local.month}月${local.day}日';
  }

  Color get color => AppColors.emotionColor(emotion);
}

File: backend/internal/fragment/domain/fragment.go
package domain

import "time"

type Fragment struct {
	ID          int64     `json:"id"`
	PublicID    string    `json:"public_id"`
	UserID      int64     `json:"user_id"`
	ContentText string    `json:"content_text"`
	Emotion     string    `json:"emotion"`
	Status      string    `json:"status"`
	Tags        []string  `json:"tags"`
	MediaURLs   []string  `json:"media_urls"`
	IsDeleted   bool      `json:"is_deleted"`
	ServerRev   int64     `json:"server_rev"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type CreateParams struct {
	ContentText string
	Emotion     string
	Tags        []string
	MediaURLs   []string
	ClientOpID  string
}

type UpdateParams struct {
	ID          int64
	ContentText string
	Emotion     string
	Tags        []string
	MediaURLs   *[]string
}

type ListQuery struct {
	Emotion string
	Tag     string
	Limit   int
}

type TimelineGroup struct {
	Label     string     `json:"label"`
	Count     int        `json:"count"`
	Fragments []Fragment `json:"fragments"`
}

type TimelineResponse struct {
	Groups  []TimelineGroup `json:"groups"`
	Items   []Fragment      `json:"items"`
	HasMore bool            `json:"has_more"`
}
